windows/RNCNetInfoCPP/RNCNetInfo.cpp

```diff
@@ -105,29 +105,43 @@ namespace winrt::ReactNativeNetInfo::implementation {
         co_return nullptr;
     }
 
-    void RNCNetInfo::Initialize(winrt::Microsoft::ReactNative::ReactContext const& /*reactContext*/) noexcept {
+    IAsyncAction RNCNetInfo::ChainGetNetworkStatus(IAsyncAction previousRequest, std::future<NetInfoState> currentRequest, std::function<void(NetInfoState)> onComplete) {

```

Comment:
no need to copy the std::function on each call here. Consider making this arg a const &

Reply:
Not sure what state the code was when you made this comment, but as it stands a copy is needed.

The reference would be to a member variable of the lambda and that lambda could be destroyed while we're waiting for one of these async operations to complete.

Reply:
fair point. We don't need a copy into the lambda and a copy into this function since its only being used in one spot though shared. Perhaps shared_ptr<std::function> so we aren't creating and tossing so many lambdas around (not that its a fantastically expensive copy with whats in it)

Reply:
Agreed, it's probably not going to be an expensive so I think going back to the copy would be best, completely forgot that we could lose the captured copy of the callback.